client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx

```diff
@@ -89,32 +85,15 @@ const getTitanMenuItems = ( {
 
 	return [
 		{
-			href: getTitanEmailUrl( titanAppsUrlPrefix, email ),
+			href: getTitanEmailUrl( titanAppsUrlPrefix, email, false, window.location.href ),
 			image: titanMailIcon,
 			imageAltText: translate( 'Titan Mail icon' ),
+			isInternalLink: true,
 			title: translate( 'View Mail', {
 				comment: 'View the Email application (i.e. the webmail) for Titan',
 			} ),
 			onClick: getTitanClickHandler( 'webmail' ),
 		},
-		{
-			href: getTitanCalendarUrl( titanAppsUrlPrefix, email ),
-			image: titanCalendarIcon,
-			imageAltText: translate( 'Titan Calendar icon' ),
-			title: translate( 'View Calendar', {
-				comment: 'View the Calendar application for Titan',
-			} ),
-			onClick: getTitanClickHandler( 'calendar' ),
-		},
-		{
-			href: getTitanContactsUrl( titanAppsUrlPrefix, email ),
```

Comment:
Same thing about this `getTitanContactsUrl()` function that can be also removed.